### ISSUE TITLE
tsconfig.json parse error manifests itself with non-obvious symptoms

### ISSUE THREAD

**Issue by uglycoyote** [Microsoft/vscode#44175]
- VSCode Version: 1.20.0
- OS Version: Mac OS

In a Typescript project using the `baseUrl` option ("Base-directory to resolve non-relative module names"),  I made edits to the `tsconfig.json` file and all of a sudden all of the imports started to fail, generating a lot of red underlines in my source files.

In this particular case, I was trying to enable the `strict` option in the TypeScript compiler and so was not terribly surprised that all of a sudden my files were full of red squigglies.  On the other hand, I was surprised when I removed `strict` from my `tsconfig.json` and things did not return to normal.  In reality, `strict` had nothing to do with the problem and it was just a general json parsing failure that led to the imports ceasing to work, but I'm including this information about the way that I experienced the bug so that you can see how VSCode's non-obvious behaviour led me astray.   

Running `tsc` on the command line there is a nice obvious message: 

`error TS5014: Failed to parse file 'tsconfig.json': Unexpected token } in JSON at position 527.`

But in Visual Studio the lack of any obvious message about the tsconfig made me assume that whatever errors I was seeing in my source files were legitimate.    It would appear that when tsconfig.json can't be parsed, VSCode falls back to parsing the Typescript as though you are using all of the default compiler options.

Rather than behaving this way, I think VSCode should

1. Make it extremely clear that `tsconfig.json` has a parse error, e.g. with one of those annoying banner messages at the top of the screen
2. Either refuse to parse your files and generate any error highlighting until you fix your `tsconfig.json` file, or parse the files using the a previous cached valid state of the `tsconfig.json` from the last time that it was able to be parsed.

**Steps to Reproduce:**

1.  Create the following files

MyProject/tsconfig.json

```
{
  "compilerOptions": {
    "baseUrl" : "src",
    "target": "ES6"
  }
}
```
MyProject/src/TestFile.ts

```
import { MyClass } from "lib/TestLibrary";

class MyTest {

    constructor() {
        let a = new MyClass();
    }

}
```

MyProject/src/lib/TestLibrary.ts

```
class MyClass {
}

export {MyClass}
```

2. Add the MyProject dir as a VSCode project directory
3. Observe that the `TestFile.ts`'s import statement resolves correctly because `tsconfig.json` has `    "baseUrl" : "src",`
4. Screw up the `tsconfig.json` by adding a trailing comma
5. Return to `Testfile.ts`. 
6. Touch the file and resave it to cause VSConfig to re-parse it
7. Observe that the import statement shows a squiggly, indicating the broken import

VSCode does mark the `tsconfig.json` file as red in the file window, indicating that there is a problem when I do this in a small sandbox project like this.  But in my real project, the tsconfig.json just shows as beige even when it has a trailing-comma parse error. (beige being the git "modified file" coloration I suppose).    So perhaps a secondary part of this bug is that that the tsconfig.json doesn't always show as red when it is unparsable.  However, in my larger project there's a lot of stuff going on in the file window,  so I'm not certain I would have noticed if the tsconfig was showing as red in there.   It's not clear to me why this fails to work properly in my production environment but I could not reproduce the bad coloring behaviour in a simple test case.

**Comment by slvganesh** [comment]
I'm facing an issue with 1.20.1 but not sure whether mine is related to this one. I have "baseUrl" configured to "c:\nodejs\node_modules" folder where all my libraries reside (such as protractor, jasmine, etc.). I'm importing the modules outside of node_modules folder. After upgrading to version 1.20.1 today, I'm started facing import issues. But this is not the case while I import modules within my node_modules folder.



Attached my tsconfig.json for reference.

**Comment by uglycoyote** [comment]
your problem is related to this issue if and only if:

* you made a syntactical error on your tsconfig file such that it was no longer value is in, such as adding an extra comma, and
* vs code did not make it clear to you that the json could not be parsed and instead the problem manifested itself in an unclear way

If you are not sure whether your tsconfig is valid json then try compiling with tsc.  It gives a clear message when the lagoon is not parsable

**Comment by mjbvz** [comment]
Partially fixed with #69880. You will see the tsconfig error in the problems tab even if you have not directly opened it:

![Screen Shot 2019-04-09 at 3 18 46 PM](https://user-images.githubusercontent.com/12821956/55838897-cb330680-5ada-11e9-8c71-0ce32b4055ae.png)

TSC now supports trailing commas too.

Real time error reporting for tsconfig files requires: https://github.com/Microsoft/TypeScript/issues/30674
